core/jazz_slack-channel/index.js

```diff
@@ -0,0 +1,504 @@
+              unknowUsers.push(usrEmail);
+            }
+          });
+
+          if (unknowUsers.length > 0) {
+            let errMsg = "Cannot find user(s) in Slack with email ids: " + unknowUsers.join(", ");
+            reject({
+              result: "inputError",
+              message: errMsg
+            });
+          } else {
+            resolve({
+              result: "success",
+              data: {
+                "registeredSlackMembers": registeredSlackMembers,
+                "channelMembers": channelMembers
+              }
+            });
+          }
+        }
+      }
+    });
+  });
+}
+
+function createPublicSlackChannel(config, eventBody) {
+  logger.debug("Inside createPublicSlackChannel");
+  return new Promise((resolve, reject) => {
+    let channel_name = eventBody.channel_name,
+    options = {
+      url: config.slack_channel_url + "channels.create?token=" + config.slack_token + "&name=" + channel_name + "&validate=true",
+      method: 'GET'
+    };
+
+    request(options, (error, response, body) => {
+      if (error) {
+        logger.error("Exception occured while creating slack channel!");
+        reject(error);
+      } else {
+        let responseBody = JSON.parse(response.body);
+        if (!responseBody || !responseBody.ok) {
+          let detailedMsg = 'Exception occured while creating slack channel: ' + channel_name;
+          logger.error('Error occured: ' + JSON.stringify(responseBody));
+          if (responseBody.error && responseBody.error === 'name_taken') {
+            detailedMsg = 'Slack channel with name: ' + channel_name + ' already exists.';
+          }
+          reject({
+            result: "inputError",
+            message: detailedMsg
+          });
+        } else {
+          let slack_link = config.slack_url + "/" + responseBody.channel.id + "/details";
+          channelInfo = {
+            'id': responseBody.channel.id,
+            'name': responseBody.channel.name,
+            'link': slack_link,
+            'members': responseBody.channel.members
+          };
+
+          resolve({
+            result: "success",
+            data: channelInfo
+          });
+        }
+      }
+    });
+  });
+}
+
+function addMemberToChannel(config, channelInfo, isServAccRequested, eventBody) {
+  logger.debug("Inside addMemberToChannel");
+  return new Promise((resolve, reject) => {
+    let slack_channel_url = config.slack_channel_url,
+    token = config.slack_token,
+    slackChannelMembers = [];
+
+    getUsersInfo(config, eventBody)
+      .then(res => {
+        let channelMembers = res.data.channelMembers;
+        let defaultMembersDetails = defaultChannelMembersDetails(res.data.registeredSlackMembers, channelInfo.members);
+        if (defaultMembersDetails.length > 0) {
+          for (let index in defaultMembersDetails) {
+            let defmember = defaultMembersDetails[index];
+            if (!isMemberAlreadyExists(defmember.id, channelMembers)) {
```

Comment:
I think you can remove the function and directly have the following:
`if (!channelMembers.find(each => each.id === defmember.id) { `

Reply:
Done